patches/tModLoader/Terraria/ModLoader/NPCLoader.cs

```diff
@@ -401,7 +401,15 @@ public static void ReceiveExtraAI(NPC npc, byte[] extraAI)
 			}
 
 			if (stream.Position < stream.Length) {
-				throw new IOException($"Read underflow {stream.Length - stream.Position} of {stream.Length} bytes in ReceiveExtraAI, more info below");
+				// Calculate original length of Read7BitEncodedInt
```

Comment:
If an under/over-read occurred in `ModNPC.ReceiveExtraAI` the contents of the `BitReader` could be complete garbage.